statsd/sender_resolving.go

```diff
@@ -78,42 +74,38 @@ func (s *ResolvingSimpleSender) Close() error {
 
 func (s *ResolvingSimpleSender) Reconnect() {
 	// lock to guard against s.running mutation
-	s.runmx.RLock()
+	s.mx.RLock()
 
 	if !s.running {
-		s.runmx.RUnlock()
+		s.mx.RUnlock()
 		return
 	}
 
-	// don't need to hold the lock while re-resolving addr
-	defer s.runmx.RUnlock()
-
 	// ro doesn't change, so no need to lock
 	ra, err := net.ResolveUDPAddr("udp", s.ro)
 	if err != nil {
 		// no good new address, so continue with old address?
 		return
 	}
 
-	// lock to guard against s.running mutation
-	s.runmx.RLock()
-	defer s.runmx.RUnlock()
+	// re-acquire lock to guard against s.running mutation after resolution
+	// (which may take a "while")
+	s.mx.RLock()
 
 	// check running again, just to be sure nothing was terminated in the meantime...
 	if s.running {
 		// only need ra write lock when mutating ra
-		s.ramx.Lock()
 		s.ra = ra
```

Comment:
You wrap `s.ra` = ra in only a RLock?

Reply:
heh. yeah, a bit of a buggy intermediate commit there. \*sigh\*